cli/dataset.go

```diff
@@ -0,0 +1,152 @@
+	}
+	resp, err := c.datasetClient.CreateDataset(context.Background(),
+		&datasetpb.CreateDatasetRequest{OrganizationId: orgID, Name: datasetName})
+	if err != nil {
+		return errors.Wrapf(err, "received error from server")
+	}
+	printf(c.c.App.Writer, "Created dataset %s with dataset ID: %s", datasetName, resp.GetId())
+	return nil
+}
+
+// DatasetRenameAction is the corresponding action for 'dataset rename'.
+func DatasetRenameAction(c *cli.Context) error {
+	client, err := newViamClient(c)
+	if err != nil {
+		return err
+	}
+	if err := client.renameDataset(c.String(datasetFlagDatasetID), c.String(datasetFlagName)); err != nil {
+		return err
+	}
+	return nil
+}
+
+// renameDataset renames an existing datasetID with the newDatasetName.
+func (c *viamClient) renameDataset(datasetID, newDatasetName string) error {
+	if err := c.ensureLoggedIn(); err != nil {
+		return err
+	}
+	_, err := c.datasetClient.RenameDataset(context.Background(),
+		&datasetpb.RenameDatasetRequest{Id: datasetID, Name: newDatasetName})
+	if err != nil {
+		return errors.Wrapf(err, "received error from server")
+	}
+	printf(c.c.App.Writer, "Dataset with ID %s renamed to %s", datasetID, newDatasetName)
+	return nil
+}
+
+// DatasetListAction is the corresponding action for 'dataset list'.
+func DatasetListAction(c *cli.Context) error {
+	client, err := newViamClient(c)
+	if err != nil {
+		return err
+	}
+	datasetIDs := c.StringSlice(datasetFlagDatasetIDs)
+	orgID := c.String(dataFlagOrgID)
+
+	if orgID != "" && datasetIDs != nil {
+		return errors.New("must specify either dataset IDs or organization ID, got both")
+	}
+	if datasetIDs != nil {
+		if err := client.listDatasetByIDs(datasetIDs); err != nil {
+			return err
+		}
+	} else {
+		if err := client.listDatasetByOrg(orgID); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// listDatasetByIDs list all datasets by ID.
+func (c *viamClient) listDatasetByIDs(datasetIDs []string) error {
+	if err := c.ensureLoggedIn(); err != nil {
+		return err
+	}
+	resp, err := c.datasetClient.ListDatasetsByIDs(context.Background(),
+		&datasetpb.ListDatasetsByIDsRequest{Ids: datasetIDs})
+	if err != nil {
+		return errors.Wrapf(err, "received error from server")
+	}
+	for _, dataset := range resp.GetDatasets() {
+		printf(c.c.App.Writer, "\t%s (ID: %s, Organization ID: %s)", dataset.Name, dataset.Id, dataset.OrganizationId)
+	}
+	return nil
+}
+
+// listDatasetByOrg list all datasets for the specified org ID.
+func (c *viamClient) listDatasetByOrg(orgID string) error {
+	if err := c.ensureLoggedIn(); err != nil {
+		return err
+	}
+	resp, err := c.datasetClient.ListDatasetsByOrganizationID(context.Background(),
+		&datasetpb.ListDatasetsByOrganizationIDRequest{OrganizationId: orgID})
+	if err != nil {
+		return errors.Wrapf(err, "received error from server")
+	}
+	for _, dataset := range resp.GetDatasets() {
+		printf(c.c.App.Writer, "\t%s (ID: %s, Organization ID: %s)", dataset.Name, dataset.Id, dataset.OrganizationId)
```

Comment:
Should this have a new line after each?

Reply:
Hm, I don't think so, when I use `list` command, `printf` actually prints each output to a new line, so `viam dataset list --org-id=0fbe951e-d4c6-427f-985f-784b7b85842c` results in

```
        alexis-dataset (ID: 65148f0831fefc36a49e5083, Organization ID: 0fbe951e-d4c6-427f-985f-784b7b85842c)
        another-test (ID: 651494d2e9548e8670754a91, Organization ID: 0fbe951e-d4c6-427f-985f-784b7b85842c)
```

where as adding a new line would result in 

```
        alexis-dataset (ID: 65148f0831fefc36a49e5083, Organization ID: 0fbe951e-d4c6-427f-985f-784b7b85842c)

        another-test (ID: 651494d2e9548e8670754a91, Organization ID: 0fbe951e-d4c6-427f-985f-784b7b85842c)

```

Reply:
Got it, didn't realize it already added a new line! Sounds good